icu4c/source/common/uloc_tag.cpp

```diff
@@ -1906,13 +1906,23 @@ ultag_parse(const char* tag, int32_t tagLen, int32_t* parsedLen, UErrorCode* sta
         return t.orphan();
     }
 
+    size_t parsedLenDelta = 0;
     /* check if the tag is grandfathered */
     for (i = 0; i < UPRV_LENGTHOF(GRANDFATHERED); i += 2) {
```

Comment:
Do we not have a hashset? Sequential comparison will be slow...

Reply:
The C++ implementation was linear search. The Java implementation use a Map. Does ICU4C internal has a Map implementation since we cannot use std::map

Reply:
We have a clunky UHashMap. If the set of keys is small, a linear search is fine...